components/playlist/browser/resources/reducers/playlist_reducer.ts

```diff
@@ -50,9 +51,21 @@ const playlistReducer: Reducer<PlaylistData | undefined> = (
       state = { ...state, currentList, lists: [...playlists] }
       break
 
-    case types.SELECTED_PLAYLIST_UPDATED:
-      const playlist = action.payload
-      state = { ...state, currentList: playlist }
+    case types.PLAYLIST_UPDATED:
+      const updatedPlaylist = action.payload
+      state = { ...state }
+      if (state.currentList?.id === updatedPlaylist.id) {
+        state.currentList = updatedPlaylist
+      }
+
+      const oldPlaylist = state.lists.find(e => {
+        return e.id === updatedPlaylist.id
+      })
+      if (oldPlaylist) {
+        state.lists[state.lists.indexOf(oldPlaylist)] = updatedPlaylist
```

Comment:
So I think this might actually not be safe in redux (even though it probably works). We're mutating `state.lists`. I think if we clone it at the top, this will be fine:


```tsx
state = { ...state, lists: [...state.lists] }
```

Internally Redux uses reference equality to determine if something has changed, so we need to make sure `lists` is a new array if we're modifying it or the change tracking (might) break

Reply:
Oh, that's good point. I learned from you as always! Thanks.